espnet2/train/preprocessor.py

```diff
@@ -1630,17 +1630,69 @@ def _speech_process(self, data: Dict[np.ndarray, str]):
 
         return data
 
+    def _convolve_rir(self, speech, rirs):
+        rir_path = np.random.choice(rirs)
+        rir = None
+        if rir_path is not None:
+            rir, _ = soundfile.read(rir_path, dtype=np.float64, always_2d=True)
+
+            # rir: (Nmic, Time)
+            rir = rir.T
+
+            # normalize rir
+            rir = rir / np.sqrt(np.sum(rir**2))
+
+            # speech: (Nmic, Time)
+            # Note that this operation doesn't change the signal length
```

Comment:
This may cause a sample shift.
Can you ensure it will not happen (or it will not be an issue)?
It may not be an issue in the small room, but it would be an issue in the medium and large rooms.

@Emrys365, how is it handled in the enhancement/separation cases?
This will cause misalignment issues in these cases.

Reply:
We either use the clean reverberant speech as target or use the early signal as target:
https://github.com/espnet/espnet/blob/master/espnet2/train/preprocessor.py#L1037-L1062

Reply:
This part has been cross-checked with https://github.com/clovaai/voxceleb_trainer/blob/343af8bc9b325a05bcf28772431b83f5b8817f5a/DatasetLoader.py#L97-L105
and the performance is also good. 

Actually, if you don't mind, I think it's okay to keep this way. Also it's only for SpkPreprocessor and I didn't change CommonPreProcessor because I wasn't too sure about the effect of this change in this setup. 

I'm not too sure about `sample shift` but I can assure that this code is working without an issue.